component_sdk/python/kfp_component/_base_op.py

```
@@ -0,0 +1,213 @@
+            return False
+        
+        try:
+            argo_execution_config = json.loads(argo_execution_config_json)
+        except Exception as e:
+            logging.error("Error deserializing argo execution config: {}".format(e))
+            return False
+        
+        deadline_json = argo_execution_config.get('deadline', None)
+        if not deadline_json:
+            return False
+        
+        try:
+            deadline = datetime.strptime(deadline_json, '%Y-%m-%dT%H:%M:%SZ')
+        except Exception as e:
+            logging.error("Error converting deadline string to datetime: {}".format(e))
+            return False
+        
+        return datetime.now() > deadline
+
+    def _stage_states(self):
+        if not self._staging_bucket or not self._staging_path or not self.staging_states:
+            return
+
+        try:
+            bucket = self._storage.get_bucket(self._staging_bucket)
+        except google.cloud.exceptions.NotFound:
+            logging.error('Unable to find staging bucket {}.'.format(self._staging_bucket))
+            return
+ 
+        blob = bucket.blob(self._staging_path)
+
+        try:
+            states_json = json.dumps(self.staging_states)
+        except TypeError as e:
+            logging.error('Unable to dump staging states: {}'.format(self.staging_states))
+            return
+
+        try:
+            blob.upload_from_string(states_json)
+        except google.cloud.exceptions.GoogleCloudError as e:
+            logging.error('Failed to upload staging states: {}'.format(e))
+            return
+
+    def _load_k8s_client(self):
+        config.load_incluster_config()
+        self._v1_core = client.CoreV1Api()
+
+    def _load_argo_metadata(self):
+        # Load argo metadata at start of an OP, as pod might be deleted in case of preemption.
+        pod = self._load_pod()
+        if not pod or not pod.metadata or not pod.metadata.labels or not pod.metadata.annotations:
+            return
```

Is it OK to silently ignore the error here?

---

I removed the staging states related code and decide to go with stateless container and move to graph component once it's ready.